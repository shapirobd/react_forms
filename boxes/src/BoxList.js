import React, { useState } from "react";
import Box from "./Box";
import NewBoxForm from "./NewBoxForm";

const BoxList = () => {
	const [boxes, setBoxes] = useState();
	return (
		<div className="BoxList">
			{boxes.map((box) => (
				<Box />
			))}
			<NewBoxForm />
		</div>
	);
};

export default BoxList;
