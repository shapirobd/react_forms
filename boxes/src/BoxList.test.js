import { render, fireEvent } from "@testing-library/react";
import BoxList from "./BoxList";

it("renders without crashing", () => {
	render(<BoxList />);
});

it("matches snapshot", () => {
	const { asFragment } = render(<BoxList />);
	expect(asFragment()).toMatchSnapshot();
});

it("should add new box to list when form submits", () => {
	const { getByLabelText, queryByText, queryByRole } = render(<BoxList />);
	const colorInput = getByLabelText("Color");
	const widthInput = getByLabelText("Width");
	const heightInput = getByLabelText("Height");
	const btn = queryByText("Create Box");

	expect(queryByRole("box")).not.toBeInTheDocument();
	fireEvent.change(colorInput, { target: { value: `black` } });
	fireEvent.change(widthInput, { target: { value: `25px` } });
	fireEvent.change(heightInput, { target: { value: `35px` } });
	fireEvent.click(btn);
	expect(queryByRole("box")).toBeInTheDocument();
});

it("should remove a box when you click on X button", () => {
	const { getByLabelText, queryByText, queryByRole } = render(<BoxList />);
	const colorInput = getByLabelText("Color");
	const widthInput = getByLabelText("Width");
	const heightInput = getByLabelText("Height");
	const submitBtn = queryByText("Create Box");

	fireEvent.change(colorInput, { target: { value: `red` } });
	fireEvent.change(widthInput, { target: { value: `50px` } });
	fireEvent.change(heightInput, { target: { value: `50px` } });
	fireEvent.click(submitBtn);
	expect(queryByRole("box")).toBeInTheDocument();

	const deleteBtn = queryByText("X");
	fireEvent.click(deleteBtn);
	expect(queryByRole("box")).not.toBeInTheDocument();
});
