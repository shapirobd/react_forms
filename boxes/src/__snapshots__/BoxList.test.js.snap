// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="BoxList"
  >
    <h3>
      Box Creator
    </h3>
    <div
      class="NewBoxForm"
    >
      <form>
        <label
          for="color"
        >
          Color
        </label>
        <input
          id="color"
          name="color"
          placeholder="Enter color"
          type="text"
          value=""
        />
        <label
          for="width"
        >
          Width
        </label>
        <input
          id="width"
          name="width"
          placeholder="Enter width"
          type="text"
          value=""
        />
        <label
          for="height"
        >
          Height
        </label>
        <input
          id="height"
          name="height"
          placeholder="Enter height"
          type="text"
          value=""
        />
        <button
          type="submit"
        >
          Create Box
        </button>
      </form>
    </div>
    <div />
  </div>
</DocumentFragment>
`;
